### PR TITLE
Adding Luis swagger files.

### PR DESCRIPTION
**Pull request by omarelhariry137**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by veronicagg** [comment]
@omarelhariry137 Please take a look at CI validation errors, semantic checks are failing - https://travis-ci.org/Azure/azure-rest-api-specs/jobs/286094532#L570

**Review by veronicagg — CHANGES_REQUESTED** [review]
Please take a look at CI, I'll post more comments soon.

**Comment by veronicagg** [comment]
@omarelhariry137 any updates on this PR fixing semantic validation failures: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/286094532 ?

**Comment by omarelhariry137** [comment]
@veronicagg Yes, they're done now. Sorry, for taking so long.

**Comment by veronicagg** [comment]
@omarelhariry137 Regarding the allowed failures section:
- mode=linter PR_ONLY=true should be as clean as possible, particularly for errors. It's green right now, though I'm not sure the tool properly ran on the files, I'll investigate. 
- mode=model PR_ONLY=true should be green - I see errors reported https://travis-ci.org/Azure/azure-rest-api-specs/jobs/289922425#L584, please take a look.

**Comment by veronicagg** [comment]
@johanste @NiklasGustafsson please let me know if you'd like to review this PR, thanks!

**Comment by veronicagg** [comment]
@omarelhariry137 Regarding linter validation, please add a file similar to https://github.com/Azure/azure-rest-api-specs/blob/current/specification/cognitiveservices/data-plane/Face/readme.md under "Luis" folder, pointing to the files included in your PR.

**Comment by veronicagg** [comment]
Here are tools you can run locally (which CI is running) https://github.com/Azure/adx-documentation-pr/wiki/Azure-Swagger-Tools

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  "paths": {
+    "/{appId}": {
+      "get": {
+        "description": "Gets the published endpoint predictions for the given query",
+        "operationId": "v2.0 Response - Get predictions from endpoint",
+        "parameters": [
+          {
+            "name": "appId",
+            "in": "path",
+            "description": "Format - guid. The application ID",
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json]: you may want to end each description with "."

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+            "name": "staging",
+            "in": "query",
+            "description": "Use staging endpoint",
+            "type": "boolean"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "A JSON object containing the predictions",
+            "examples": {
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json]: you could look into using x-ms-examples - https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/x-ms-examples.md#why-x-ms-examples
It allows multiple examples and you can put the example in a separate file.

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+        "produces": [
+          "application/json"
+        ]
+      },
+      "post": {
+        "description": "Gets the published endpoint prediction for the given long query",
+        "operationId": "v2.0 Response - Get predictions from endpoint (post)",
+        "parameters": [
+          {
+            "name": "appId",
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json]: if parameters are shared among operations you could define them in global section and refer them from here. 
https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#parameters-definitions-object
Make sure to include https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-parameter-location as needed.

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json (hunk 2 above)
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json]: should there be a schema definition returned?
similar to https://github.com/Azure/azure-rest-api-specs/blob/current/specification/cognitiveservices/data-plane/Face/v1.0/Face.json#L52

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json [hunk 4]
```diff
@@ -0,0 +1,5225 @@
...
+    },
+    {
+      "apiKeyQuery": []
+    }
+  ],
+  "paths": {
+    "/apps/{appId}/versions/{versionId}/patterns": {
+      "post": {
+        "description": "Creates a new pattern feature.",
+        "operationId": "features - Create pattern feature",
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json]: operationID should be of a form similar to https://github.com/Azure/azure-rest-api-specs/blob/current/specification/cognitiveservices/data-plane/Face/v1.0/Face.json#L31 
The string before the underscore "_" groups the operations in a class named that way (in code generation)
Something like "Features_CreatePattern"
The name is used for the generated method name.

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json [hunk 5]
```diff
@@ -0,0 +1,5225 @@
...
+          "application/json"
+        ],
+        "responses": {
+          "201": {
+            "description": "The ID of the created feature.",
+            "examples": {
+              "application/json": "83147"
+            }
+          },
+          "400": {
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json]: Are you intending to throw and exception in generated code? Please see https://github.com/Azure/azure-rest-api-specs/blob/2fb9a0b3b902335ff0b0033711c234431931ec9d/documentation/creating-swagger.md#default-response on how autorest acts on default response and negative responses

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+    },
+    {
+      "apiKeyQuery": []
+    }
+  ],
+  "paths": {
+    "/{appId}": {
+      "get": {
+        "description": "Gets the published endpoint predictions for the given query",
+        "operationId": "v2.0 Response - Get predictions from endpoint",
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Endpoint API.swagger.json]: operationID should be of a form similar to https://github.com/Azure/azure-rest-api-specs/blob/current/specification/cognitiveservices/data-plane/Face/v1.0/Face.json#L31
The string before the underscore "_" groups the operations in a class named that way (in code generation)
Something like "Predictions_Get"
The name is used for the generated method name.

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json [hunk 7]
```diff
@@ -0,0 +1,5225 @@
...
+            "name": "phraselistId",
+            "in": "path",
+            "description": "The ID of the feature to be deleted.",
+            "required": true,
+            "type": "integer"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": ""
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json]: description should not be empty

File: specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json [hunk 8]
```diff
@@ -0,0 +1,5225 @@
...
+              "name": "TakeParameter",
+              "modelAsString": false
+            },
+            "enum": [
+              100
+            ]
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (veronicagg) [inline specification/cognitiveservices/data-plane/Luis/v2.0/LUIS Programmatic APIs v2.0.swagger.json]: is this a pageable operaation? see https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable

**Review by veronicagg — CHANGES_REQUESTED** [review]
Left a bunch of comments inline - many of them apply to all operations, I commented once, so you can review all. 
I'd also recommend generating SDK code using autorest https://github.com/Azure/autorest/blob/master/docs/user/cli.md#autorest-command-line-interface-documentation so you can preview what your SDK would look like.

**Comment by veronicagg** [comment]
@omarelhariry137 any updates on the PR? The PR will get closed by EOD due to inactivity. If so, you can reopen it and tag me when you get back to it.

**Pull request closed without merging**